internal/cli/cli.go

```diff
@@ -12,14 +12,17 @@ import (
 	"github.com/pelletier/go-toml/v2"
 )
 
-type ConvertFn func(r io.Reader, w io.Writer) error
+type Options map[string]interface{}
```

Comment:
For now I'd like this to be a `struct`, to lean on go's default values to avoid exists/empty checks and casting, and make it easier to understand which options are supported.

Because `Program` is used by multiple programs, a single `Options` struct may be unlikely to work with all of them. I think the easiest way around this is to have `convert()` take an options struct, and return a function, creating a closure around the provided options. That way each program can build its own options an generate its own processing function, without the need for a one-size-fits-all options type.